keras_cv/models/object_detection/retina_net/retina_net.py

```diff
@@ -516,19 +528,13 @@ def compute_metrics(self, x, y, y_pred, sample_weight):
 
     @classmethod
     def from_config(cls, config):
-        config["backbone"] = keras.utils.deserialize_keras_object(
-            config["backbone"]
-        )
         return super().from_config(config)
 
     def get_config(self):
         return {
             "num_classes": self.num_classes,
             "bounding_box_format": self.bounding_box_format,
-            "backbone": keras.utils.serialize_keras_object(self.backbone),
-            # TODO(haifengj): handle custom anchor_generator. we now rely on
-            # label_encoder and prediction_decoder to reconstruct
-            # anchor_gnerator.
+            "feature_extractor": self.feature_extractor,
```

Comment:
Why is superior to serializing the `backbone`? We serialize the backbone without issue in `ImageClassifier`

Reply:
the problem is that some `trainable_variables` on `backbone` are unused.  So you get all these annoying warnings telling you that you didn't pass a `loss` to compile(), when in fact the variables are just not used.

Reply:
I'll fix this in a separate PR - this way I can fix one thing at a time and make it easier to review.

Reply:
I've reverted, we can address in a separate PR.